File: ui-api/src/main/java/io/tipsters/uiapi/repository/CompetitionRepository.java
package io.tipsters.uiapi.repository;

import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.Query;
import org.springframework.data.repository.query.Param;
import org.springframework.stereotype.Repository;

import java.util.List;
import java.util.Set;
import java.util.UUID;

import io.tipsters.uiapi.domain.Competition;

@Repository
public interface CompetitionRepository extends JpaRepository<Competition, UUID> {

  @Query(nativeQuery = true, value = "select c.name from competitions c where cast(c.id as text) in :ids")
  Set<String> findCompetitionNamesByIdIn(@Param("ids") List<String> ids);
}

File: web-server/src/main/kotlin/io/tipsters/repository/CompetitionRepository.kt
package io.tipsters.repository

import io.tipsters.domain.Competition
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.data.jpa.repository.Query
import org.springframework.data.repository.query.Param
import org.springframework.stereotype.Repository
import java.util.*

@Repository
interface CompetitionRepository : JpaRepository<Competition, UUID> {
    @Query(nativeQuery = true, value = "select c.name from competitions c where cast(c.id as text) in :ids")
    fun findCompetitionNamesByIdIn(@Param("ids") ids: List<String>): Set<String>
}
